services/worker-manager/test/provider_azure_test.js

```diff
@@ -634,11 +628,46 @@ helper.secrets.mockSuite(testing.suiteName(), [], function(mock, skipping) {
       await assertProvisioningState({ ip: 'allocated', nic: 'allocated', vm: 'none' });
       assert(provider.removeWorker.called);
     });
+  });
 
-    test('successful provisioning of VM without public ip', async function() {
-      await assertProvisioningState({ ip: 'none', nic: 'none', vm: 'none' });
+  suite('provisionResources without public IP', function() {
```

Comment:
extracted this test in separate case, as its success depends on initial `setup()`, worker pool needs to have proper `workerConfig` to set `skipPublicIP` properly